### ISSUE TITLE
Provide deterministic API

### ISSUE THREAD

**Issue by koraa** [PQClean/PQClean#428]
It would be useful if pqclean exposed deterministic versions of KEMs and Signature Algorithms taking explicit coins/secret seed parameters. These would come in handy because some protocols mandate their use [0] and they would provide protocol authors with the ability to easily arrive at and check test vectors for their protocols.

[0]: https://eprint.iacr.org/2020/379

**Comment by thomwiggers** [comment]
I agree completely that this would be extremely useful. It would also have avoided all the messiness that we've had to build around the different implementations that we have of `randombytes()`.

However, the reality is that the NIST API hasn't provided this, so offering this isn't trivial and would requiring significant effort to modify every single scheme.
